src/storage/txn/mod.rs

```diff
@@ -56,6 +56,17 @@ pub enum ProcessResult {
     },
 }
 
+impl ProcessResult {
+    pub fn maybe_clone(&self) -> Option<ProcessResult> {
```

Comment:
It's not a proper name because it either returns a `Some` or panics. How about this?
```rust
impl ProcessResult {
    pub fn maybe_clone(&self) -> Option<ProcessResult> {
        match self {
            ProcessResult::PessimisticLockRes { res: Ok(r) } => {
                Some(ProcessResult::PessimisticLockRes { res: Ok(r.clone()) })
            }
            _ => None,
        }
    }
}
```
And caller must be in charge of the result.

Reply:
Addressed.
On the caller, unwrap() should be enough I think